packages/server/modules/core/repositories/streams.ts

```diff
@@ -1018,14 +1018,18 @@ export async function markCommitStreamUpdated(commitId: string) {
 // TODO: Replace all calls to `grantStreamPermissions` with this
 export const upsertProjectRoleFactory =
   ({ db }: { db: Knex }): UpsertProjectRole =>
-  async ({ projectId, userId, role }) => {
+  async (
+    { projectId, userId, role },
+    { trackProjectUpdate } = { trackProjectUpdate: true }
```

Comment:
I'd rather the option is `preventProjectUpdate`, cause now all implementations have to remember to make it default to true, and the same kind of cognitive load is also attached to anyone using those implementations (the dev has to check the default value). Would reduce confusion IMO.

Other than that, looks fine